cdc/kv/client.go

```diff
@@ -1059,10 +1035,12 @@ func (s *eventFeedSession) handleError(ctx context.Context, errInfo regionErrorI
 	case *sendRequestToStoreErr:
 		metricStoreSendRequestErr.Inc()
 	default:
+		//[TODO] Move all OnSendFail logic here
+		// We expect some unknown error to trigger RegionCache recheck its store state and change leader to peer to
+		// make some detection(peer may tell us where new leader is)
+		// RegionCache.OnSendFail is thread_safe inner.
 		bo := tikv.NewBackoffer(ctx, tikvRequestMaxBackoff)
-		if errInfo.rpcCtx.Meta != nil {
-			s.regionCache.OnSendFail(bo, errInfo.rpcCtx, regionScheduleReload, err)
-		}
```

Comment:
How can we guarantee `errInfo.rpcCtx.Meta` is not nil?

Reply:
1. errInfo.rpcCtx.Meta is the origin Region object we get from pd, it can not be nil if we always keep rpcCtx as we get from RegionCache.
2. It matter nothing if errInfo.rpcCtx.Meta is nil, because RegionCache.OnSendFail will get *Region from inner cache using RegionVerID. 
3. 'errInfo.rpcCtx.Meta != nil' is related to PartialClone, which I guest is to avoid leader switch(just try), but I think it's unnecessary.